packages/core/src/account/smartContractAccount.ts

```diff
@@ -88,7 +88,7 @@ export type ToSmartContractAccountParams<
   getAccountInitCode: () => Promise<Hex>;
   getDummySignature: () => Hex;
   encodeExecute: (tx: Tx) => Promise<Hex>;
-  encodeBatchExecute: (txs: Tx[]) => Promise<Hex>;
+  encodeBatchExecute?: (txs: Tx[]) => Promise<Hex>;
   // if not provided, will default to just using signMessage over the Hex
   signUserOperationHash?: (uoHash: Hex) => Promise<Hex>;
   encodeUpgradeToAndCall?: (params: UpgradeToAndCallParams) => Promise<Hex>;
```

Comment:
one q:   why is `encodeUpgradeToAndCall` in core smart contract account? isn't think only relevant to light account contract?